src/preprocessor.rs

```diff
@@ -400,6 +450,103 @@ mod test {
+                // here we chop the big string into up to 4 bits
+                let mut chopped = Vec::with_capacity(4);
+                let mut giant_string: String = tuple.1.clone();
+                while giant_string.len() > 0 && chopped.len() < 4 {
+                    // verify we are at a char boundary
+                    let indices = giant_string.char_indices();
+                    let num_chars = giant_string.chars().count();
+                    if let Some((index, _)) = indices.skip(num_chars / 2).next() {
+                        let mut splitted = giant_string.split_off(index);
+                        std::mem::swap(&mut splitted, &mut giant_string);
+                        chopped.push(splitted);
+                    } else {
+                        break;
+                    }
+                }
+                chopped.push(giant_string);
+                (tuple.0, chopped)
+            })
+            .boxed()
+    }
+
+    proptest! {
+        #[test]
+        fn textual_length_prefix_prop((lengths, datas) in multiple_textual_lengths(5)) {
+            let mut pre_p = pre::TextualLength::default();
+            let mut in_ns = 0_u64;
+            let res: Vec<Vec<u8>> = datas.into_iter().map(|data| {
+                pre_p.process(&mut in_ns, data.as_bytes()).unwrap()
+            }).flatten().collect();
+            assert_eq!(lengths.len(), res.len());
+            for (processed, expected_len) in res.iter().zip(lengths) {
+                assert_eq!(expected_len, processed.len());
+            }
+        }
+    }
+
+    #[test]
+    fn textual_prefix_length_loop() {
+        let datas = vec![
+            "24 \'?\u{d617e}ѨR\u{202e}\u{f8f7c}\u{ede29}\u{ac784}36 ?{¥?MȺ\r\u{bac41}9\u{5bbbb}\r\u{1c46c}\u{4ba79}¥\u{7f}*?:\u{0}$i",
+            "60 %\u{a825a}\u{a4269}\u{39e0c}\u{b3e21}<ì\u{f6c20}ѨÛ`HW\u{9523f}V",
+            "\u{3}\u{605fe}%Fq\u{89b5e}\u{93780}Q3",
+            "¥?\u{feff}9",
+            " \'�2\u{4269b}",
+        ];
+        let lengths: Vec<usize> = vec![24, 36, 60, 9];
+        let mut pre_p = pre::TextualLength::default();
+        let mut in_ns = 0_u64;
+        let res: Vec<Vec<u8>> = datas
+            .into_iter()
+            .map(|data| pre_p.process(&mut in_ns, data.as_bytes()).unwrap())
+            .flatten()
+            .collect();
+        assert_eq!(lengths.len(), res.len());
+        for (processed, expected_len) in res.iter().zip(lengths) {
+            assert_eq!(expected_len, processed.len());
+        }
+    }
+
+    #[test]
+    fn textual_length_prefix() {
+        let mut pre_p = pre::TextualLength::default();
+        let data = textual_prefix(42);
+        let mut in_ns = 0_u64;
+        let mut res = pre_p.process(&mut in_ns, data.as_bytes()).unwrap();
+        assert_eq!(1, res.len());
+        let payload = res.pop().unwrap();
+        assert_eq!(42, payload.len());
+    }
+
+    #[test]
+    fn empty_textual_prefix() {
+        let data = ("").as_bytes();
```

Comment:
Seeing this, i also thought about an empty textual-prefix-length, like so `0 `. This should be valid, right? You think this is worth adding a test too?